workers/handshake-manager/src/manager/matching/internal_engine.rs

```diff
@@ -39,86 +41,43 @@ impl HandshakeExecutor {
     ///        denormalization easier
     pub async fn run_internal_matching_engine(
         &self,
-        order: OrderIdentifier,
+        order_id: OrderIdentifier,
     ) -> Result<(), HandshakeManagerError> {
-        info!("Running internal matching engine on order {order}");
+        info!("Running internal matching engine on order {order_id}");
         // Lookup the order and its wallet
-        let (network_order, wallet) = self.fetch_order_and_wallet(&order).await?;
+        let (network_order, wallet) = self.fetch_order_and_wallet(&order_id).await?;
         let my_order = wallet
             .orders
             .get(&network_order.id)
+            .cloned()
             .ok_or_else(|| HandshakeManagerError::State(ERR_NO_ORDER.to_string()))?;
-        let (my_proof, my_witness) = self
-            .get_validity_proof_and_witness(&network_order.id)
-            .await?
-            .ok_or_else(|| HandshakeManagerError::State(ERR_MISSING_PROOFS.to_string()))?;
+        let sell_mint = my_order.send_mint();
+        let my_balance = wallet.get_balance_or_default(sell_mint);
```

Comment:
1. Looks like there's no `wallet.get_balance_or_default` method, so I guess you've added it in another PR, but at any rate
2. This block of code looks logically equivalent to your new `get_managed_order_and_balance` method, can use it here